ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java

```diff
@@ -87,6 +87,39 @@ public static Integer compute_shuffled_index(int index, int index_count, Bytes32
     return indexRet;
   }
 
+  /**
+   * Return from ``indices`` a random index sampled by effective balance.
+   *
+   * @param state
+   * @param indices
+   * @param seed
+   * @return
+   */
+  public static int compute_proposer_index(BeaconState state, List<Integer> indices, Bytes32 seed) {
+    checkArgument(!indices.isEmpty(), "compute_proposer_index indices must not be empty");
+    UnsignedLong MAX_RANDOM_BYTE = UnsignedLong.valueOf(255); // Math.pow(2, 8) - 1;
+    int i = 0;
+    while (true) {
+      int candidate_index =
+          indices.get(compute_shuffled_index(i % indices.size(), indices.size(), seed));
+      int random_byte =
+          UnsignedBytes.toInt(
+              Hash.sha2_256(Bytes.concatenate(seed, int_to_bytes(Math.floorDiv(i, 32), 8)))
```

Comment:
We only need to regenerate the hash when when `i%32 == 0` :D

Reply:
Good spot. That seems like a worthwhile optimisation to pick up now - I was translating the spec pretty blindly.